### PR TITLE
Add blog post building an app with Vanilla JS

### PR DESCRIPTION
**Pull request by mraible**

### CODE REVIEW

File: _source/_posts/2018-06-05-authentication-vanilla-js.md [hunk 1]
```diff
@@ -0,0 +1,1043 @@
...
+}
+```
+
+There are other ways you can remove nodes with the DOM like `parent.removeChild(node)` but for the sake of simplicity, `node.remove()` tends to be the easiest to understand.
+
+### Render Lists
+
+It's common for web apps to render items inside a loop.  Since you will be doing this in your app for rendering individual meals, I wanted to step you through a few ways you can accomplish this and showcase some hidden drawbacks to each approach.
+
+**Example 1:** Concat each list element to the parents `innerHTML` property:
```
Comment (aaronpk) [inline _source/_posts/2018-06-05-authentication-vanilla-js.md]: "parent's"

File: _source/_posts/2018-06-05-authentication-vanilla-js.md [hunk 2]
```diff
@@ -0,0 +1,1043 @@
...
+  /* ... */
+  for (let meal of meals) {
+    let elMeal = document.createElement('div')
+    elMeal.innerHTML = `<span>${meal.title}</span>`
+    el.appendChild(elMeal)
+  }
+}
+```
+
+By using `appendChild()` we can achieve this which cuts down significantly on the processing. But, there is one minor drawback to this approach when rendering larger lists: reflows. Reflow is the technical term of the web browser process that computes the layout of the page. For every loop iteration, the browser has to recalculate the position of each appended element.
```
Comment (aaronpk) [inline _source/_posts/2018-06-05-authentication-vanilla-js.md]: Replace first sentence: "we cut down significantly..."

File: _source/_posts/2018-06-05-authentication-vanilla-js.md [hunk 3]
```diff
@@ -0,0 +1,1043 @@
...
+    }
+  })
+}
+```
+
+### Use Async and Await
+
+With ES6, `async/await` brings you added control over asynchronous programming by leveraging promises but in a synchronous way. You can use `async/await` to make your code cleaner while still maintaining non-blocking operations.
+
+To illustrate this, I'll briefly show you how the evolution of asynchronous programming.
```
Comment (aaronpk) [inline _source/_posts/2018-06-05-authentication-vanilla-js.md]: Delete "how"

File: _source/_posts/2018-06-05-authentication-vanilla-js.md (hunk 2 above)
Comment (mraible) [reply _source/_posts/2018-06-05-authentication-vanilla-js.md]: I'm gonna go with: "Using `appendChild()` cuts down significantly on the processing."

**Pull request merged**